Lib/glyphsLib/builder/builders.py

```diff
@@ -387,17 +387,37 @@ def _valid_designspace(self, designspace):
         """Make sure that the user-provided designspace has loaded fonts and
         that names are the same as those from the UFOs.
         """
-        # TODO: (jany) really make a copy to avoid modifying the original object
-        copy = designspace
-        for source in copy.sources:
+        import copy
+
+        designspace_copy = copy.copy(designspace)
+
+        sources = []
+        sources_to_rename = []
+        for source in designspace_copy.sources:
+            if source.layerName:
+                if not ("{" in source.layerName and "}" in source.layerName):
+                    sources_to_rename.append(source)
+                    self.logger.warning(
+                        "Source {source_name}, layerName '{layer_name}': "
+                        "layer name will be renamed to Glyphs.app brace layer "
+                        "conventions.".format(
+                            source_name=source.filename, layer_name=source.layerName
+                        )
+                    )
+            else:
+                sources.append(source)
+        designspace_copy.sources = sources
+
+        filename_to_ufo = {}
+        for source in designspace_copy.sources:
             if not hasattr(source, "font") or source.font is None:
                 if source.path:
-                    # FIXME: (jany) consider not changing the caller's objects
                     source.font = defcon.Font(source.path)
```

Comment:
@madig you removed the FIXME comment, but note that you're still modifying the original SourceDescriptor objects, because you're only doing a shallow `copy.copy`. But I think it's ok here.